api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -368,12 +368,37 @@ public Workspace findByWorkspaceId(long workspaceId) {
       Map<String, WorkspaceAccessEntry> rolesMap) {
     List<UserRole> userRoles = new ArrayList<>();
     for (Map.Entry<String, WorkspaceAccessEntry> entry : rolesMap.entrySet()) {
-      User user = userDao.findUserByEmail(entry.getKey());
-      userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      // Filter out groups
+      try {
+        User user = userDao.findUserByEmail(entry.getKey());
+        userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      } catch (NullPointerException e) {
+        log.log(Level.WARNING, "No user found for " + entry.getKey());
+      }
     }
     return userRoles.stream()
         .sorted(
             Comparator.comparing(UserRole::getRole).thenComparing(UserRole::getEmail).reversed())
         .collect(Collectors.toList());
   }
+
+  public Workspace doWorkspacePublish(
+      Workspace workspace, String publishedWorkspaceGroup, boolean publish) {
+    Map<String, WorkspaceAccessEntry> firecloudAcls =
+        getFirecloudWorkspaceAcls(workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
+
+    Map<String, WorkspaceAccessLevel> updatedAcls = new HashMap<>();
+    for (Map.Entry<String, WorkspaceAccessEntry> entry : firecloudAcls.entrySet()) {
+      updatedAcls.put(
+          entry.getKey(), WorkspaceAccessLevel.fromValue(entry.getValue().getAccessLevel()));
+    }
+
+    if (publish) {
+      updatedAcls.put(publishedWorkspaceGroup, WorkspaceAccessLevel.READER);
+    } else {
+      updatedAcls.put(publishedWorkspaceGroup, WorkspaceAccessLevel.NO_ACCESS);
+    }
+
+    return updateWorkspaceAcls(workspace, updatedAcls);
```

Comment:
Where does this update the ACLs in FC? Presumably at some point we need to make a post call?

Reply:
`updateWorkspaceAcls` makes the firecloud call to update on the FC end